plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/mapping/RuleBasedIdentifierMapping.java

```diff
@@ -83,28 +82,28 @@ public String fromRemoteColumnName(String remoteColumnName)
     }
 
     @Override
-    public String toRemoteSchemaName(ConnectorIdentity identity, Connection connection, String schemaName)
+    public String toRemoteSchemaName(ConnectorIdentity identity, String schemaName, RemoteSchemaNamesProvider remoteSchemasProvider, boolean shouldUpperCase)
     {
         String remoteSchemaName = toRemoteSchema.get(schemaName);
         if (remoteSchemaName == null) {
-            remoteSchemaName = delegate.toRemoteSchemaName(identity, connection, schemaName);
+            remoteSchemaName = delegate.toRemoteSchemaName(identity, schemaName, remoteSchemasProvider, shouldUpperCase);
         }
         return remoteSchemaName;
     }
 
     @Override
-    public String toRemoteTableName(ConnectorIdentity identity, Connection connection, String remoteSchema, String tableName)
+    public String toRemoteTableName(ConnectorIdentity identity, String remoteSchema, String tableName, RemoteTableNamesProvider remoteTablesProvider, boolean shouldUpperCase)
     {
         String remoteTableName = toRemoteTable.get(remoteSchema, tableName);
         if (remoteTableName == null) {
-            remoteTableName = delegate.toRemoteTableName(identity, connection, remoteSchema, tableName);
+            remoteTableName = delegate.toRemoteTableName(identity, remoteSchema, tableName, remoteTablesProvider, shouldUpperCase);
         }
         return remoteTableName;
     }
 
     @Override
-    public String toRemoteColumnName(Connection connection, String columnName)
+    public String toRemoteColumnName(String columnName, boolean shouldUpperCase)
     {
-        return delegate.toRemoteColumnName(connection, columnName);
+        return delegate.toRemoteColumnName(columnName, shouldUpperCase);
     }
 }
```

Comment:
```
public interface IdentifierMapping
{
    String fromRemoteSchemaName(String remoteSchemaName);

    String fromRemoteTableName(String remoteSchemaName, String remoteTableName);

    String fromRemoteColumnName(String remoteColumnName);

    String toRemoteSchemaName(RemoteIdentifiers remoteSchemasProvider, ConnectorIdentity identity, String schemaName);

    String toRemoteTableName(RemoteIdentifiers remoteIdentifiers, ConnectorIdentity identity, String remoteSchema, String tableName);

    String toRemoteColumnName(RemoteIdentifiers remoteIdentifiers, String columnName);
}

public interface RemoteIdentifiers
{
    Set<String> getRemoteSchemas();

    Set<String> getRemoteTables(String remoteSchema);

    boolean shouldUpperCase();
}
```

Then for most JDBC connector you can use same class for `RemoteIdentifiers` and different one for Druid.